ui/app/src/Views/AlertList/AlertView.elm

```diff
@@ -0,0 +1,91 @@
+module Views.AlertList.AlertView exposing (view)
+
+import Alerts.Types exposing (Alert)
+import Html exposing (..)
+import Html.Attributes exposing (class, style, href)
+import Html.Events exposing (onClick)
+import Types exposing (Msg(CreateSilenceFromAlert, Noop, MsgForAlertList))
+import Utils.Date
+import Views.AlertList.Types exposing (AlertListMsg(AddFilterMatcher))
+import Utils.Views exposing (buttonLink)
+import Utils.Filter
+import Time exposing (Time)
+
+
+view : Alert -> Html Msg
+view alert =
+    li
+        [ class "align-items-center list-group-item alert-list-item p-0 d-inline-flex justify-content-start"
+        ]
+        [ dateView alert.startsAt
+        , labelButtons alert.labels
+        , div [ class "ml-auto d-inline-flex align-self-stretch p-2", style [ ( "border-left", "1px solid #ccc" ) ] ]
+            [ generatorUrlButton alert.generatorUrl
+            , silenceButton alert
+            ]
+        ]
+
+
+dateView : Time -> Html Msg
+dateView time =
+    i
+        [ class "h-100  d-flex flex-column justify-content-center p-2 text-muted"
+        , style [ ( "border-right", "1px solid #ccc" ), ( "font-family", "monospace" ) ]
+        ]
+        [ span [] [ text <| Utils.Date.timeFormat time ]
+        , small [] [ text <| Utils.Date.dateFormat time ]
+        ]
+
+
+labelButtons : List ( String, String ) -> Html Msg
+labelButtons labels =
+    let
+        -- the alertname label should be first
+        sortedLabels =
+            List.append
```

Comment:
Looks like a good case for [`List.partition`](http://package.elm-lang.org/packages/elm-lang/core/latest/List#partition). Together with [`uncurry`](http://package.elm-lang.org/packages/elm-lang/core/5.1.1/Basics#uncurry) this looks nice and short:

```
labelButtons labels =
  labels
  |> List.partition (Tuple.first >> (==) "alertname") -- returns a tuple of two lists
  |> uncurry (++) -- joins two lists from a tuple
  |> List.map labelButton -- creates a list of lables
  |> div [] -- wraps labels in a div
```

You can then rewrite it in a point free style, but I wouldn't do it because it decreases readability:

```
labelButtons =
 List.partition (Tuple.first >> (==) "alertname")
 >> uncurry (++)
 >> List.map labelButton
 >> div []
```